frontend/src/diagram/DiagramWebSocketContainer.tsx

```diff
@@ -550,32 +551,47 @@ export const DiagramWebSocketContainer = ({
    * initialization will be done each time we are in the loading state.
    */
   useEffect(() => {
-    const onSelectElement = (newSelectedElement, diagramServer) => {
+    const onSelectElement = (selectedElement: SModelElement, diagramServer: DiagramServer) => {
       const newSelection: Selection = { entries: [] };
-      if (newSelectedElement.root.id === newSelectedElement.id) {
-        // newSelectedElement => SGraph
-        const { id, label, kind } = newSelectedElement;
+
+      if (selectedElement instanceof SGraph) {
+        const { id, label, kind } = selectedElement as any; // (as any) to be removed when the proper type will be available
         newSelection.entries.push({ id, label, kind });
-      } else {
-        // newSelectedElement => SNode || SEdge
-        const { targetObjectId, targetObjectKind, targetObjectLabel } = newSelectedElement;
-        newSelection.entries.push({
+      } else if (selectedElement instanceof SNode || selectedElement instanceof SEdge) {
+        const { id, targetObjectId, targetObjectKind, targetObjectLabel } = selectedElement as any; // (as any) to be removed when the proper type will be available
+
+        const semanticSelectionEntry: SelectionEntry = {
           id: targetObjectId,
           label: targetObjectLabel,
           kind: targetObjectKind,
-        });
+        };
+
+        const type = selectedElement instanceof Node ? 'Node' : 'Edge';
```

Comment:
Looking inside the state, the selected graphical element is always an `Edge` like if _selectedElement_ was never a `Node`